packages/atomic/src/components/common/items-per-page/validate.ts

```diff
@@ -0,0 +1,32 @@
+import {AtomicCommerceProductsPerPage} from '../../commerce/atomic-commerce-products-per-page/atomic-commerce-products-per-page';
+import {AtomicResultsPerPage} from '../../search/atomic-results-per-page/atomic-results-per-page';
+
+export function validateChoicesDisplayed(
+  context: AtomicResultsPerPage | AtomicCommerceProductsPerPage
+) {
+  return context.choicesDisplayed.split(',').map((choice) => {
+    const parsedChoice = parseInt(choice);
+    if (isNaN(parsedChoice)) {
+      const errorMsg = `The choice value "${choice}" from the "choicesDisplayed" option is not a number.`;
+      context.bindings.engine.logger.error(errorMsg, context);
+      throw new Error(errorMsg);
+    }
+
+    return parsedChoice;
+  });
+}
+
+export function validateInitialChoice(
+  context: AtomicResultsPerPage | AtomicCommerceProductsPerPage,
+  choices: number[]
+) {
+  if (!context.initialChoice) {
+    context.initialChoice = choices[0];
```

Comment:
I don't think the function should have a side effect like that. Instead, it should return a value that indicates what the initial choice will be.

Then, we can remove the `mutable: true` property on the `initialChoice` prop

Reply:
Thanks, instead I removed the logic from the validate function and added this just before instead

`      this.initialChoice = this.initialChoice ?? this.choices[0];`